app/src/main/java/org/oppia/android/app/profile/ProfileChooserFragmentPresenter.kt

```diff
@@ -286,11 +286,27 @@ class ProfileChooserFragmentPresenter @Inject constructor(
     }
   }
 
-  private fun ensureProfileOnboarded(profile: Profile) {
-    if (profile.profileType == ProfileType.SUPERVISOR || profile.completedProfileOnboarding) {
-      logInToProfile(profile)
-    } else {
+  private fun computeLoginRoute(profile: Profile) {
+    when {
+      profile.profileType == ProfileType.SUPERVISOR -> { routeToHomeOrPinScreens(profile) }
+      !profile.completedProfileOnboarding -> { routeToOnboardingOrPinScreens(profile) }
```

Comment:
```suggestion
      profile.startedProfileOnboarding -> { routeToOnboardingOrPinScreens(profile) }
```

Is likely more explicit

Reply:
This actually means a different thing in this scenario. Migrated profiles will nevers have profile.startedProfileOnboarding == true, so keeping the original.